ivadomed/loader/slice_filter.py

```diff
@@ -40,24 +47,24 @@ def __init__(self, filter_empty_mask=True,
             else:
                 self.classifier = torch.load(classifier_path, map_location='cpu')
 
-    def __call__(self, sample):
+    def __call__(self, sample: dict):
         input_data, gt_data = sample['input'], sample['gt']
+        """Extract input_data and gt_data lists from sample dict and discard them if they don't match certain
+        conditions.
 
+        """
         if self.filter_empty_mask:
-            # Filter slices that do not have ANY ground truth (i.e. all masks are empty)
+            # Discard slices that do not have ANY ground truth (i.e. all masks are empty)
             if not np.any(gt_data):
                 return False
-
         if self.filter_absent_class:
```

Comment:
Actually you raise an excellent point about using "discard". Do you think `self.discard_empty_mask` would be a better flag? It certainly convey the message more clear than `filter` as filter can be both filter in or filter out.

Up to your preference. Just thought about it per your earlier change.

Reply:
- I agree, we could go with `discard_empty_something` instead of `filter_empty_something` for both `PatchFilter` and `SliceFilter`.
- `PatchFilter` and `SliceFilter` themselves would need to be renamed... `PatchDiscard` and `SliceDiscard`?
- Then we need to account for retrocompatibility by changing the keys from old config files to the new keys in the KEY_CHANGE_DICT in the config manager [here](https://github.com/ivadomed/ivadomed/blob/76b36a0a0f7141feb2d5b00b33e4c3a06865fc2c/ivadomed/config_manager.py#L71).

We could either:
1. Do another PR to deal with renaming SliceFilter and retrocompatiblity that would go in before this PR, then update this PR with PatchDiscard from issue Also in this issue: https://github.com/ivadomed/ivadomed/issues/997.
2. Merge this PR first and rename afterwards.

I prefer option 1 because it's cleaner to make the change in SliceFilter before implementing PatchFilter, but it may take more time to finally get this new feature in.

At this point, I think we just need to take a decision about the name :)
More discussion about the naming in a comment above https://github.com/ivadomed/ivadomed/pull/980#issuecomment-1000535086

I could try to have a PR ready for review for the renaming by the next meeting so we can move on.
Let me know what you think.

Reply:
Hi @dyt811,
Thank you again for the review on this complicated PR.
I discussed twith @jcohenadad  and we agreed to merge this PR and take care of the renaming/refactoring afterwards with issue #997.